src/annotator/components/ProfileModal.tsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useRef, useState } from 'preact/hooks';
+import classnames from 'classnames';
+import { CancelIcon, IconButton } from '@hypothesis/frontend-shared/lib/next';
+
+import type { Emitter, EventBus } from '../util/emitter';
+
+export type ProfileConfig = { profileAppUrl: string } & Record<string, unknown>;
+
+type ProfileModalProps = {
+  eventBus: EventBus;
+  config: ProfileConfig;
+};
+
+export function ProfileModal({ eventBus, config }: ProfileModalProps) {
+  const [isHidden, setIsHidden] = useState(true);
+  const emitterRef = useRef<Emitter | null>(null);
+
+  useEffect(() => {
+    const emitter = eventBus.createEmitter();
+    emitter.subscribe('openProfile', () => {
+      setIsHidden(false);
+    });
+    emitterRef.current = emitter;
+
+    return () => {
+      emitter.destroy();
+    };
+  }, [eventBus]);
+
+  const onClose = () => {
+    setIsHidden(true);
+    emitterRef.current?.publish('closeProfile');
+  };
+
+  return (
+    <div
+      className={classnames(
+        'fixed z-max top-0 left-0 right-0 bottom-0 p-3 bg-black/50',
+        { hidden: isHidden }
+      )}
+      data-testid="profile-outer"
+    >
+      <div className="relative w-full h-full" data-testid="profile-inner">
+        <div className="absolute right-0 m-3">
+          <IconButton
+            title="Close the Profile"
+            onClick={onClose}
+            variant="dark"
+            classes={classnames(
+              // Remove the dark variant's background color to avoid
+              // interfering with modal overlays. Re-activate the dark variant's
+              // background color on hover.
+              // See https://github.com/hypothesis/client/issues/3676
+              '!bg-transparent enabled:hover:!bg-grey-3'
+            )}
+          >
+            <CancelIcon className="w-4 h-4" />
+          </IconButton>
```

Comment:
In my first iteration, this component was very close to `NotebookModal`. After the suggestions from https://github.com/hypothesis/client/pull/5214#discussion_r1094547417, they are not that close anymore so maybe it makes sense to keep the small duplications for simplicity.

The only thing I think we can safely extract is the close button, as suggested here https://github.com/hypothesis/client/pull/5214#discussion_r1094545481

I'll do it once this series of PRs is merged.